.github/workflows/app.yml

```diff
@@ -0,0 +1,76 @@
+name: "All the sdk specific jobs"
+
+on:
+  workflow_call:
+    inputs:
+      flutter_version:
+        description: "The flutter version used for tests and builds"
+        type: string
+        required: true
+      dart_version:
+        description: "The dart version used for tests and builds"
+        type: string
+        required: true
+
+jobs:
+  e2ee_test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        homeserver: [synapse, dendrite, conduit]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run tests
+        run: |
+          export NETWORK='--network mynet'
+          docker network create mynet
+          # deploy homeserver instance
+          scripts/integration-server-${{matrix.homeserver}}.sh
+          docker run $NETWORK --env GITHUB_ACTIONS="${GITHUB_ACTIONS}" --env HOMESERVER_IMPLEMENTATION="${{matrix.homeserver}}" --env HOMESERVER="${{startsWith('dendrite', matrix.homeserver) && format('{0}:8008', matrix.homeserver) || matrix.homeserver }}" --volume="$(pwd):/workdir" --workdir /workdir ghcr.io/famedly/container-image-flutter/flutter:${{inputs.flutter_version}} /bin/bash -c "set -e
+          scripts/integration-prepare-alpine.sh
+          # create test user environment variables
+          source scripts/integration-create-environment-variables.sh
+          # properly set the homeserver IP and create test users
+          scripts/integration-prepare-homeserver.sh 
+          # setup OLM
+          scripts/prepare.sh
+          scripts/test_driver.sh"
+
+  coverage:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/famedly/container-image-flutter/flutter:${{inputs.flutter_version}}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run tests
+        run: |
+          sed -i 's/#\([^ ]\)/\1/g' pubspec.yaml
```

Comment:
isn't so safe, touches dependency_overrides too for example. Could just make it `sed -i 's/#flutter_test/flutter_test/g' pubspec.yaml` is much simpler and readable